modules/openapi-generator/src/main/resources/powershell-experimental/rsa_provider.mustache

```diff
@@ -0,0 +1,377 @@
+            }
+            //showBytes("Final key material", keymaterial);
+            byte[] deskey = new byte[24];
+            Array.Copy(keymaterial, deskey, deskey.Length);
+
+            Array.Clear(psbytes, 0, psbytes.Length);
+            Array.Clear(data00, 0, data00.Length);
+            Array.Clear(result, 0, result.Length);
+            Array.Clear(hashtarget, 0, hashtarget.Length);
+            Array.Clear(keymaterial, 0, keymaterial.Length);
+
+            return deskey;
+        }
+
+        public static string GetRSASignb64encode(string private_key_path, byte[] digest)
+        {
+            RSACryptoServiceProvider cipher = new RSACryptoServiceProvider();
+            cipher = GetRSAProviderFromPemFile(private_key_path);
+            RSAPKCS1SignatureFormatter RSAFormatter = new RSAPKCS1SignatureFormatter(cipher);
+            RSAFormatter.SetHashAlgorithm("SHA256");
+            byte[] signedHash = RSAFormatter.CreateSignature(digest);
+            return Convert.ToBase64String(signedHash);
+        }
+
+        public static RSACryptoServiceProvider GetRSAProviderFromPemFile(String pemfile)
+        {
+            bool isPrivateKeyFile = true;
+            if (!File.Exists(pemfile))
+            {
+                throw new Exception("pemfile does not exist.");
+            }
+            string pemstr = File.ReadAllText(pemfile).Trim();
+            if (pemstr.StartsWith(pempubheader) && pemstr.EndsWith(pempubfooter))
+                isPrivateKeyFile = false;
+
+            byte[] pemkey = null;
+            if (isPrivateKeyFile)
+                pemkey = DecodeOpenSSLPrivateKey(pemstr);
+
+
+            if (pemkey == null)
+                return null;
+
+            if (isPrivateKeyFile)
+            {
+                return DecodeRSAPrivateKey(pemkey);
+            }
+            return null;
+        }
+
+        static byte[] DecryptKey(byte[] cipherData, byte[] desKey, byte[] IV)
+        {
+            MemoryStream memst = new MemoryStream();
+            TripleDES alg = TripleDES.Create();
+            alg.Key = desKey;
+            alg.IV = IV;
+            try
+            {
+                CryptoStream cs = new CryptoStream(memst, alg.CreateDecryptor(), CryptoStreamMode.Write);
+                cs.Write(cipherData, 0, cipherData.Length);
+                cs.Close();
+            }
+            catch (Exception exc)
+            {
+                Console.WriteLine(exc.Message);
+                return null;
+            }
+            byte[] decryptedData = memst.ToArray();
+            return decryptedData;
+        }
+
+        static SecureString GetSecPswd(String prompt)
+        {
+            SecureString password = new SecureString();
+
+            Console.ForegroundColor = ConsoleColor.Gray;
+            Console.ForegroundColor = ConsoleColor.Magenta;
+
+            while (true)
+            {
+                ConsoleKeyInfo cki = Console.ReadKey(true);
```

Comment:
The generated code should not force the client to use interactive input through the console. All input must be provided as configuration parameters, and it's up to the client to decide how to get these parameters (e.g. from console, from a secret store, etc)

Reply:
removed the console for password input, value can be provided using method argument